great_expectations/rule_based_profiler/domain_builder/domain_builder.py

```
@@ -1,53 +1,111 @@
+        """
+
+        if data_context is None:
+            raise ge_exceptions.ProfilerExecutionError(
+                message=f"{self.__class__.__name__} requires a data_context, but none was provided."
+            )
 
-        Note: In this class, we do not verify that all of these batch_ids are accessible; this should be done elsewhere
-        (with an error raised in the appropriate situations).
+        self._data_context = data_context
+        self._batch_request = batch_request
 
+    def get_domains(
+        self,
+        variables: Optional[ParameterContainer] = None,
+    ) -> List[Domain]:
+        """
         Note: Please do not overwrite the public "get_domains()" method.  If a child class needs to check parameters,
         then please do so in its implementation of the (private) "_get_domains()" method, or in a utility method.
         """
-        return self._get_domains(
-            validator=validator,
-        )
+        return self._get_domains(variables=variables)
 
     @abstractmethod
     def _get_domains(
         self,
-        *,
-        validator: Optional[Validator] = None,
+        variables: Optional[ParameterContainer] = None,
     ) -> List[Domain]:
         """
         _get_domains is the primary workhorse for the DomainBuilder
+        """
+
+        pass
 
-        IMPORTANT: If an implementation of DomainBuilder sets "batch_id": my_batch_id in "Domain.domain_kwargs" and also
-        calls "validator.get_metric()" in its logic, then "MetricConfiguration" must also set "batch_id" as follows:
-        validator.get_metric(
-            metric=MetricConfiguration(
-                metric_name="my_metric",
-                metric_domain_kwargs={key_mdkw0: value_mdkw0, key_mdkw1: value_mdkw1, "batch_id": my_batch_id,},
-                metric_value_kwargs={key_mvkw0: value_mvkw0, key_mvkw1: value_mvkw1,},
-                metric_dependencies={key_mdeps0: value_mdeps0, key_mdeps1: value_mdeps1,},
+    def get_batch_id(
```

Given that this method will raise an error if there is more than one batch in the `BatchRequest`, I wonder if we should move it to the child classes. I can imagine a future scenario where we have a MultiBatchDomainBuilder, and so we wouldn't want to raise an error in this case. Thoughts?

---

@talagluck In general, I do not disagree.  Moreover, I made a small but useful refactor of this class as part of another PR (will come after this one is merged).  How about we reassess at that time?  It should be relatively straightforward.  Thanks.

---

Sounds reasonable!